tests/fixtures/test_executable_mismatch_resources/exe001_neg.py

```diff
@@ -0,0 +1,3 @@
+if __name__ == '__main__':
```

Comment:
I don't like the idea to store these files as fixtures.
We can use `tempfile` instead: https://docs.pytest.org/en/latest/tmpdir.html

That's how I see it:
1. We store fixtures as strings inside a text file
2. We store file modes as parametrize values there as well
3. We use `tempfile` and file mode to set permissions before testing
4. We test with temp files without having real ones

Does this sound reasonable to you?